### DISCUSSION TITLE
Cylindrical lamination for slotless motor

### DISCUSSION THREAD

**Discussion by SupaDupaTwo** [Eomys/pyleecan#618] [General]
Hi, 

I 'd like to add slotless motor in pyleecan but I face several issue. Has someone found a way to do it?
I thought using Slot type 24 and setting W3 to 0. Geometrically it is correct but the boundary conditions are not applied on the edge of the coil and the lamination (since the lam tooth where it should be applied is not there). 
I tried creating a cylindrical lam model but the coil are no boundary condition applied on coils

![slotless](https://user-images.githubusercontent.com/122090534/236845240-23148bda-0972-45ee-bb3c-c24993477857.png)

**Comment by SupaDupaTwo** [comment]
btw I found an walkaround by removing the symmetry manually  (but at the cost of increased calculation time).

**Comment by BonneelP** [comment]
Hello,

In pyleecan normally there is no difference between a slot for magnet and a slot for winding . The slot you are looking for should then be SlotM18:
![SlotM18](https://user-images.githubusercontent.com/20533034/237021946-0ea1bfcd-13a0-4261-ac1d-d00dac6817d9.png)
This slot was added especially to handle the boundary conditions (it is a particular case of SlotM11) so hopefully it should work (I have never tested it with winding instead of magnets).

Best regards,
Pierre

**Reply by SupaDupaTwo** [reply]
Thanks for the suggestion. we are almost there but there is still an issue how to draw the airgap boundary (highlighted in red) 
![Capture d’écran 2023-05-09 143532](https://github.com/Eomys/pyleecan/assets/122090534/a9617acc-fa14-49d3-9fd2-3040814e1a0b)
![Capture d’écran 2023-05-09 143440](https://github.com/Eomys/pyleecan/assets/122090534/5357287b-5ac0-417c-937e-898a5c52129f)
![Capture d’écran 2023-05-09 143505](https://github.com/Eomys/pyleecan/assets/122090534/62b59a06-57e9-46f5-a398-6a2169b2953b)

**Comment by BonneelP** [comment]
I think that the issue is the lamination definition. 
As the slot is thought to be used for a magnet, I guess that you need to update the stator.Rint value to match that:
![image](https://github.com/Eomys/pyleecan/assets/20533034/9fe1fd46-5da6-40ee-a276-57365f8b598d)

**Reply by SupaDupaTwo** [reply]
Rint is matching the lamination internat radius. The problem, I think, is that it doesn't take into account the coil. It sees a large air gap and places the label and airgap boundary at locations where the coils should be located.
changing rint to match the coil inner results solve the airgap postion but the coil is then wrong  
![Capture d’écran 2023-05-09 154147](https://github.com/Eomys/pyleecan/assets/122090534/e125ec1c-a3a8-4401-af4d-5c395ad43c0d)

**Reply by BonneelP** [reply]
For "machine with magnets" the conventions for Lamination radius are:
![Dimension_Stator_Rotor_mag](https://github.com/Eomys/pyleecan/assets/20533034/0fdf3421-aa2f-46d1-b2f6-880b24b98d8f)
![Dimension_Rotor_Stator_mag](https://github.com/Eomys/pyleecan/assets/20533034/02e718aa-b389-4883-8ccf-c05d778bc060)

By checking your first screen shot the reference point of your lamination and winding seems to be correct so I guess that it's an error in the code: I guess that it is in the LamSlot.comp_radius_mec methods that are used to define the airgap width. As most of the time the winding is inside a slot, we assume that the mechanical radius is Rint which is not the case here (the issue is handle for LamSlotMag but not LamSlotWind). Do you think you can take a look in the code (I don't know when I will have the time to investigate/solve that).

Best regards,
Pierre

**Reply by SupaDupaTwo** [reply]
in pyleecan\Functions\FEMM\get_sliding_band.py, there are geometrical definition of the airgap should I try to add it here?

```
def get_sliding_band(sym, lam_int, lam_ext):
    """Returns  a list of surface in the airgap including the sliding band surface

    Parameters
    ----------
    sym: int
        Symmetry factor (1= full machine, 2= half of the machine...)
    Rgap_mec_int: float
        Internal lamination mechanic radius
    Rgap_mec_ext: float
        External lamination mechanic radius

    Returns
    -------
    surf_list: list
        List of surface in the airgap including the sliding band surface
    """

    Rgap_mec_int, Rgap_mec_ext = lam_int.comp_radius_mec(), lam_ext.comp_radius_mec()
    Rgap_0_int, Rgap_0_ext = _get_radius_boundary(sym, lam_int, lam_ext)
    Wgap_mec = Rgap_mec_ext - Rgap_mec_int
    W_sb = Wgap_mec / 3  # Width sliding band
    surf_list = list()
    label_int = lam_int.get_label()
    label_ext = lam_ext.get_label()

    if sym == 1:  # Complete machine
        # Bottom sliding band
        surf_list.append(
            Circle(
                center=0,
                radius=Rgap_mec_int + W_sb,
                label=label_int + "_" + SLID_LAB,
                point_ref=(Rgap_mec_int + W_sb / 2) * exp(1j * pi / 2),
                prop_dict={BOUNDARY_PROP_LAB: SBR_B_LAB},  # Bottom Radius
            )
        )
        # Top sliding band
        surf_list.append(
            Circle(
                center=0,
                radius=Rgap_mec_int + 2 * W_sb,
                label=label_ext + "_" + SLID_LAB,
                point_ref=(Rgap_mec_ext - W_sb / 2) * exp(1j * pi / 2),
                prop_dict={BOUNDARY_PROP_LAB: SBR_T_LAB},  # Top Radius
            )
        )
        # Middle
        surf_list.append(
            SurfLine(
                line_list=[],
                point_ref=(Rgap_mec_int + W_sb * 3 / 2) * exp(1j * pi / 2),
                label=NO_LAM_LAB + "_" + NO_MESH_LAB,
            )
        )
    else:  # Symmetry
        # Bottom line
        Z1 = Rgap_0_int
        Z2 = Rgap_mec_int + W_sb
        Z3 = Z2 * exp(1j * 2 * pi / sym)
        Z4 = Z1 * exp(1j * 2 * pi / sym)
        airgap_lines = list()
        airgap_lines.append(  # Bottom Right side
            Segment(begin=Z1, end=Z2, prop_dict={BOUNDARY_PROP_LAB: SBS_BR_LAB})
        )
        airgap_lines.append(
            Arc1(
                begin=Z2,
                end=Z3,
                radius=Rgap_mec_int + W_sb,  # Bottom Radius
                prop_dict={BOUNDARY_PROP_LAB: SBR_B_LAB},
            )
        )
        airgap_lines.append(  # Bottom Left side
            Segment(begin=Z3, end=Z4, prop_dict={BOUNDARY_PROP_LAB: SBS_BL_LAB})
        )
        surf_list.append(
            SurfLine(
                line_list=airgap_lines,
                point_ref=(Z2 - W_sb / 2) * exp(1j * pi / sym),
                label=label_int + "_" + SLID_LAB,
            )
        )
        # Top line
        Z5 = Rgap_0_ext
        Z6 = Rgap_mec_ext - W_sb
        Z7 = Z6 * exp(1j * 2 * pi / sym)
        Z8 = Z5 * exp(1j * 2 * pi / sym)
        airgap_lines = list()
        airgap_lines.append(  # Top Right Side
            Segment(begin=Z5, end=Z6, prop_dict={BOUNDARY_PROP_LAB: SBS_TR_LAB})
        )
        airgap_lines.append(
            Arc1(  # Top Radius
                begin=Z6,
                end=Z7,
                radius=Rgap_mec_ext - W_sb,
                prop_dict={BOUNDARY_PROP_LAB: SBR_T_LAB},
            )
        )
        airgap_lines.append(  # Top Left Side
            Segment(begin=Z7, end=Z8, prop_dict={BOUNDARY_PROP_LAB: SBS_TL_LAB})
        )
        surf_list.append(
            SurfLine(
                line_list=airgap_lines,
                point_ref=(Z6 + W_sb / 2) * exp(1j * pi / sym),
                label=label_ext + "_" + SLID_LAB,
            )
        )

    return surf_list
```

**Reply by BonneelP** [reply]
I just checked and it's more complexe that I first thought :/ 
There is also a Boundary condition trick that is done only for LamSlotMag that needs to be transferred to the LamSlotWind method. I will try to solve that this week. Can you share your machine so I can create a dedicated test ?

**Reply by SupaDupaTwo** [reply]
Here it is : [slotess_M18.zip](https://github.com/Eomys/pyleecan/files/11443107/slotess_M18.zip) 
Thanks

**Reply by BonneelP** [reply]
Hello, 

I managed to solve the issue with PR #621
It helps me find a bug with the Boundary conditions for SlotM18 even in the magnet case :)

I have added a test on your machine to make sure that this kind of topology can be simulated by Pyleecan. Can I add a screenshot of your machine to the website gallery ? https://pyleecan.org/gallery.html

I'm running again all the tests one last time before merging the PR into master. Will you need a pip release afterwards ?

Best regards,
Pierre

**Reply by SupaDupaTwo** [reply]
Nice, I pulled your changes and it works! Many thanks. 
I can wait the next release for get the pip release and yes you can add this in the gallery.
David
